### PR TITLE
Remove fixture files from repo and instead generate them from code

### PR DESCRIPTION
**Pull request by jmccann**
Trying to remove fixtures directory and generate what is needed from code.

### CODE REVIEW

File: archive/tar/tar_test.go [hunk 1]
```diff
@@ -131,18 +134,93 @@ func unpackIt(a archive.Archive, src string) error {
 	return a.Unpack("/dev/null", reader)
 }
 
+func createGoodTarfile() {
```
Comment (donny-dont) [inline archive/tar/tar_test.go]: I'm a bit confused as to why this is here. Shouldn't it just write the files and then the tar code creates the tar.

**Review by donny-dont — CHANGES_REQUESTED** [review]
So here's what I imagined for doing stuff without fixtures

For writing
* Setup would create the files in a temp directory somewhere.
* Tar file would be written
* Examine the contents of it through a reader

For reading
* Take the previous tar file and expand it
* Make sure everything is written out the same way

**Comment by jmccann** [comment]
Thanks for the direction.  Will get it implemented.

**Comment by jmccann** [comment]
So here's what I implemented

For writing
* Setup creates the files in a temp directory somewhere.
* Tar file would be written

For reading
* Take the previous tar file and expand it
* Make sure everything is written out the same way

Still missing cause I'm not sure how
* Examine the tar file contents through a reader.

Let me know you thoughts on what I did so far.  Also was wondering if you could help on the missing item or if we could just skip it for now since Unpack checks sort of verify that part as well (though it doesn't keep the testing for things in their own set of tests I admit).

**Review by donny-dont — APPROVED** [review]

**Pull request merged**